src/_pytest/python.py

```diff
@@ -1190,14 +1190,25 @@ def idmaker(argnames, parametersets, idfn=None, ids=None, config=None, item=None
         _idvalset(valindex, parameterset, argnames, idfn, ids, config=config, item=item)
         for valindex, parameterset in enumerate(parametersets)
     ]
-    if len(set(ids)) != len(ids):
-        # The ids are not unique
-        duplicates = [testid for testid in ids if ids.count(testid) > 1]
-        counters = Counter()
+
+    # All ids must be unique!
+    unique_ids = set(ids)
+    if len(unique_ids) != len(ids):
+
+        # Record the number of occurrences of each test id
+        testid_counts = Counter(ids)
+
+        # Map the test id to its next suffix.
+        testid_suffixes = Counter(unique_ids)
```

Comment:
as no counter api is used for that value, i suggest to use a `defaultdict(int)` here to avoid the 2 extra loops

Reply:
Thanks! Resolved in d040385